src/classes/WalletConnectors/WalletConnectConnector.ts

```diff
@@ -119,6 +133,16 @@ export class WalletConnectConnector extends WalletConnector {
         await this.setChain(chainId)
     }
 
+    public async disconnect() {
+        try {
+            await this.provider.disconnect()
+        }
+        catch (e) {
+            if (!(e as Error).message.includes('No matching key'))
+                throw e
+        }
```

Comment:
maybe checking if the wallet is connected before calling disconnect will be a better approach.

this catch block doesn't explicit to the reader why you're checking if the message contains 'No matching key'.

Reply:
Im not entirely sure what `provider.disconnect()` does under the hood... Also, i would probably not check if it is connected before executing the method since Wallet Connect stores some data on local storage and this probably cleans some of this stuff to avoid unexpected errors with invalid / expired sessions...

I could add a comment about that inside the catch block, but as far as I know that's how it should be done...

Reply:
Okay, a comment explaining the above works then